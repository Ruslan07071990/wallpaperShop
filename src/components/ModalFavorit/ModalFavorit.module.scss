.modalWrapper {
    position: fixed;
    top: 0;
    left: 0; 
    right: 0;
    bottom: 0;
    background-color: rgba(49, 46, 46, 1);
    z-index: 1;
    display: flex;
    justify-content: center;
    align-items: center;
}

.modal {
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    max-width: 720px;
    height: 340px;
    box-sizing: border-box;
    background-color: #e74c3c;
    border-radius: 4px;
    color: mediumorchid;


    .modalContent {
        display: flex;
        flex-direction: column;
        justify-content: flex-start;
        align-items: center;
        font-family: Roboto,sans-serif;
        font-weight: 400;
        color: #fff;

        .modalClose {
            position: absolute;
            right: 28px;
            top: 0;
            font-size: 60px;
            font-weight: 100;
            display: inline-flex;
          
            text-align: center;
            box-sizing: border-box;
            align-items: center;
            justify-content: center;
           
            border: none;
            color: #fff;
            cursor: pointer;

          }

        .modalHeader {
            width: 720px;
            height: 85px;
            display: flex;
            justify-content: space-between;
            align-items: center;
            background-color: #d44637;
            
            & h2 {
                margin: 0 30px;
                font-size: 32px;
                
            }
        }
        .bodyText {
            margin: 30px;
            line-height: 1.6;
            font-size: 30px;
        }
        .modalActions{
            font-size: 30px;
        }
    }
}