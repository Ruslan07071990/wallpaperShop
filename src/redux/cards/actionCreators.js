import { SET_CARD } from "./actions";
import axios from 'axios';



export const setCards = (cards) => ({ type: SET_CARD, payload: cards });


export const axiosCards = () => {
    return async (dispatch) => {
        try {
            const { data } = await axios.get('./products.json');
            dispatch(setCards(data));

        } catch (error) {
            console.log('ERROR', error);
        }
    }
}
