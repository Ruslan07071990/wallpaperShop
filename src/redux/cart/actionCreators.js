import { SET_CART, ADD_TO_CART, DELETE_FROM_CART } from "./actions";
import { CART_LS_KEY } from "../../components/constants";
import { getStateFromLocalStorage } from "../../components/utils/localStorageHelper";

export const setCart = (cards) => ({
  type: SET_CART,
  payload: cards,
});

export const addToCart = (cards) => ({
  type: ADD_TO_CART,
  payload: cards,
});

export const deleteFromCart = (cards) => ({
  type: DELETE_FROM_CART,
  payload: cards,
});

export const axiosCart = () => {
  return async (dispatch) => {
    try {
      const cartFromLs = getStateFromLocalStorage(CART_LS_KEY);

      if (cartFromLs) {
        dispatch(setCart(cartFromLs));
      }

    } catch (error) {
      console.log("ERROR", error);
    }
  };
};
