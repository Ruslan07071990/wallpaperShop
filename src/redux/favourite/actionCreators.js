import { SET_FAVOURITE, CHANGE_FAVOURITE } from "./actions";
import { FAV_LS_KEY } from "../../components/constants";
import { getStateFromLocalStorage } from "../../components/utils/localStorageHelper";

export const setFavourite = (cards) => ({
  type: SET_FAVOURITE,
  payload: cards,
});

export const changeFavourite = (cards) => ({
  type: CHANGE_FAVOURITE,
  payload: cards,
});

export const axiosFavourite = () => {
  return async (dispatch) => {
    try {
      const favFromLs = getStateFromLocalStorage(FAV_LS_KEY);

      if (favFromLs) {
        dispatch(setFavourite(favFromLs));
      }

    } catch (error) {
      console.log("ERROR", error);
    }
  };
};
