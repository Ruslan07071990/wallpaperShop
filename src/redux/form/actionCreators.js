import { FORM_CHEC } from "./actions";



export const formBuyChec = (value) => ({
    type: FORM_CHEC,
    payload: value,
  });