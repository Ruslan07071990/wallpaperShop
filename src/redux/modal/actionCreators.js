import { SET_MODAL, SET_MODAL_ACTION, SET_MODAL_TITLE,PRODUKT_PRICE,PRODUKT_NAME } from "./actions";

export const setModal = () => ({ type: SET_MODAL });

export const setModalAction = (value) => ({
  type: SET_MODAL_ACTION,
  payload: value,
});
export const setModalTitle = (value) => ({
  type: SET_MODAL_TITLE,
  payload: value,
});

export const setProductName = (value) => ({
  type: PRODUKT_NAME,
  payload: value,
});
export const setProductPrice = (value) => ({
  type: PRODUKT_PRICE,
  payload: value,
});

