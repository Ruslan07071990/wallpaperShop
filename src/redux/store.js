import {createStore, applyMiddleware} from "redux"
import appReduser from "./appRedusers";
import { composeWithDevTools } from "redux-devtools-extension";
import thunk from "redux-thunk";

const store = createStore(appReduser,composeWithDevTools(applyMiddleware(thunk)));
export default store;