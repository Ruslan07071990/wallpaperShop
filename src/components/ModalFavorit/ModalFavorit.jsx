import React from "react";
import styles from "./ModalFavorit.module.scss";
import PropTypes from "prop-types";
import { useSelector, useDispatch } from "react-redux";
import { setModal } from "../../redux/modal/actionCreators";

const ModalFavorite = ({ closeButton }) => {
  const btnConfirm = useSelector((state) => state.modal.modalAction);
  const productName = useSelector((state) => state.modal.name);
  const productPrice = useSelector((state) => state.modal.price);
  const titleModal = useSelector((state) => state.modal.title);

  const dispatch = useDispatch();
  const toggleModal = () => {
    dispatch(setModal());
  };
  const fonnClick = (event) => {
    // Перевіряємо, чи клік був здійснений за межами модального вікна
    if (event.currentTarget === event.target) {
      //Якщо так, то додаємо код для закриття модального вікна
      toggleModal();
    }
  };
  return (
    <div className={styles.modalWrapper} onClick={fonnClick}>
      <div className={styles.modal}>
        <div className={styles.modalContent}>
          <div className={styles.modalHeader}>
            <h2>{titleModal}</h2>

            {closeButton && (
              <span className={styles.modalClose} onClick={toggleModal}>
                &times;
              </span>
            )}
          </div>
          <div>
            <p className={styles.bodyText}></p>
            <span className={styles.modalActions}>
              {productName} - {productPrice} грн
            </span>
          </div>
          <div className={styles.modalActions}>{btnConfirm}</div>
        </div>
      </div>
    </div>
  );
};
ModalFavorite.defaultProps = {
  closeButton: true,
};

ModalFavorite.propTypec = {
  closeModal: PropTypes.bool,
};
export default ModalFavorite;
