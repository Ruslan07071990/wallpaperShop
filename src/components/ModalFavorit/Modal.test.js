import { render } from "@testing-library/react";
import ModalFavorit from './ModalFavorit'
import { Provider, useDispatch } from "react-redux";
import store from "../../redux/store";
import { useEffect } from "react";
import { setModal } from "../../redux/modal/actionCreators";

const Dispatcher = () => {
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(setModal(true));
  }, []);

  return <button onClick={() => dispatch(setModal(true))}>OPEN MODAL</button>;
};

const Component = (props) => {
  return (
    <Provider store={store}>
      <Dispatcher />
      <ModalFavorit />
    </Provider>
  );
};

describe("Snapshot test", () => {
  test("should Modal render", () => {
    const { asFragment } = render(<Component />);

    expect(asFragment()).toMatchSnapshot();
  });
});
