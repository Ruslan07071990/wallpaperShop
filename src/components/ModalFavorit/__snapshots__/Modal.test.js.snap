// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test should Modal render 1`] = `
<DocumentFragment>
  <button>
    OPEN MODAL
  </button>
  <div
    class="modalWrapper"
  >
    <div
      class="modal"
    >
      <div
        class="modalContent"
      >
        <div
          class="modalHeader"
        >
          <h2 />
          <span
            class="modalClose"
          >
            ×
          </span>
        </div>
        <div>
          <p
            class="bodyText"
          />
          <span
            class="modalActions"
          >
             -  грн
          </span>
        </div>
        <div
          class="modalActions"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
